### PR TITLE
Implement the Required Changes

### PR DESCRIPTION
**Pull request by IsmailChangezi**
## Description.
- Use React Routers 
- Use Redux for state Management
- Use API to extract Data
- Render the data from the API
- Style the page
- Fix Linter errors
-

### CODE REVIEW

File: src/components/Currency.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+export default function Currency(data) {
+  const { currency, bg } = data;
+  const {
+    id, rank, name, symbol, priceUsd,
+  } = currency;
+  // console.log(currency);
+  /* const dispatch = useDispatch();
+  const currencyDetails = (currency) => {
+    dispatch(showDetails(currency));
+  }; */
```
Comment (GedeonTS) [inline src/components/Currency.js:18]: - As I requested above, Kindly make sure to remove commented out lines of code to make you code clean and professional :recycle:

File: src/pages/Details.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { NavLink, useParams } from 'react-router-dom';
+import { FaChevronLeft, FaMicrophone, FaCog } from 'react-icons/fa';
+// import store from '../redux/configureStore';
+// import { getCrypto } from '../redux/currencies/currencies';
+
```
Comment (GedeonTS) [inline src/pages/Details.js:7]: - As I requested above, Kindly make sure to remove commented out lines of code to make you code clean and professional :recycle:

File: src/__tests__/__snapshots__/Currency.test.js.snap [hunk 3]
```diff
@@ -0,0 +1,28 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (GedeonTS) [inline src/__tests__/__snapshots__/Currency.test.js.snap:1]: - Kindly make sure all your tests are passing, as there are some which are failing :point_down: 
Screenshot from 2023-04-27 14-52-21

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @IsmailChangezi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [ ] Nice job with the project :rocket: 


 ## Required Changes ♻️
_Check the comments under the review._
- [ ] It would be better if we resolve the horizontal scroll at the bottom of the app as we can see in the screenshot below
Screenshot from 2023-04-27 14-15-44
- [ ] Also we are missing the `Acknowledgements` section which we should give credit to the author of the original design (Nelson Sakawa). This is also a case for invalidation
- [ ] It would be better if we ensure all failing tests are fixed before submitting another code review request
Screenshot from 2023-04-27 14-20-01



## Optional suggestions
- [ ] It is very important to follow the project requirements, here we did not follow the correct Git-flow work flow, thus creating the PR from a feature branch, for example; `crypto_test` into the `dev` branch but instead we followed the GitHub-workflow which shouldn't be the case and should have been invalidated by the first reviewer
Screenshot from 2023-04-27 13-55-11

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**